File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/RibbonBar/Appearance/AddingGroupsPage.cs
using System;
using System.Text;
using System.Data;
using System.Drawing;
using System.ComponentModel;
using System.Collections.Generic;

using Gizmox.WebGUI.Forms;
using Gizmox.WebGUI.Common;

namespace CompanionKit.Controls.RibbonBar.Appearance
{
    public partial class AddingGroupsPage : UserControl
    {
        public AddingGroupsPage()
        {
            InitializeComponent();

			demoRibbonBar.SelectedIndex = 0;
			SelectedIndexChanged(demoRibbonBar, new EventArgs());
        }

		/// <summary>
		/// Handle click on "advanced" anchor
		/// </summary>
		private void AdvancedClicked(object sender, RibbonBarGroupArgs e)
		{
			if (e != null && e.Group != null)
			{
				lblAdvanced.Visible = true;
				lblAdvanced.Text = string.Format("The advanced anchor: {0}", e.Group.Text);
			}
		}

		/// <summary>
		/// Handle RibbonBarPage activation
		/// </summary>
		private void SelectedIndexChanged(object sender, EventArgs e)
		{
			if (demoRibbonBar.SelectedIndex > -1)
			{
				RibbonBarPage objPage = demoRibbonBar.Pages[demoRibbonBar.SelectedIndex];
				if (objPage != null)
				{
					lblStatus.Text = string.Format("The active page: {0}", objPage.Text);
				}
			}
		}
    }
}

File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/RibbonBar/Appearance/AddingGroupsPage.vb
Namespace CompanionKit.Controls.RibbonBar.Appearance

	Public Class AddingGroupsPage
		Inherits UserControl
		' Methods
		Public Sub New()
			Me.InitializeComponent()
			Me.demoRibbonBar.SelectedIndex = 0
			Me.SelectedIndexChanged(Me.demoRibbonBar, New EventArgs)
		End Sub

		Private Sub AdvancedClicked(ByVal sender As Object, ByVal e As RibbonBarGroupArgs)
			If ((Not e Is Nothing) AndAlso (Not e.Group Is Nothing)) Then
				Me.lblAdvanced.Visible = True
				Me.lblAdvanced.Text = String.Format("The advanced anchor: {0}", e.Group.Text)
			End If
		End Sub

		Private Sub SelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs)
			If (Me.demoRibbonBar.SelectedIndex > -1) Then
				Dim objPage As RibbonBarPage = Me.demoRibbonBar.Pages.Item(Me.demoRibbonBar.SelectedIndex)
				If (Not objPage Is Nothing) Then
					Me.lblStatus.Text = String.Format("The active page: {0}", objPage.Text)
				End If
			End If
		End Sub
	End Class

End Namespace

